contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/security/WithMockJwt.java

```diff
@@ -0,0 +1,63 @@
+package com.contentgrid.spring.test.security;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.annotation.AliasFor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+import org.springframework.security.test.context.support.WithSecurityContext;
+import org.springframework.security.test.context.support.WithSecurityContextFactory;
+import org.springframework.util.StringUtils;
+
+@Target({ElementType.METHOD, ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Inherited
+@Documented
+@WithSecurityContext(factory = WithMockJwt.WithMockJwtSecurityContextFactory.class)
+@Import({SkipCsrfConfiguration.class})
```

Comment:
Now I'm wondering what happens with this import on the method-level ? :thinking:
That won't work, right ?

Reply:
Removing the `@WithMockJwt` on the class and placing it on the method testing POST -> test fails with http 403

Reply:
If it's not supported on methods, then `@Target` should not include `ElementType.METHOD`

Reply:
but it is intended to be used on method level too ...

Reply:
Setting the `@WithMockJwt` on the method overrides the properties of the one on the class. It is when the `@WithMockJwt` is missing on the class that the `SkipCsrfConfiguration` does not get loaded.

Reply:
In that case, you might need to use `@Import(SkipCsrfConfiguration.class)` on the test-class first. 

Ok, fine with me.